test_communication/CMakeLists.txt

```diff
@@ -164,7 +164,11 @@ if(BUILD_TESTING)
     if(rmw_implementation2 MATCHES "(.*)fastrtps(.*)")
       set(rmw_implementation2_is_fastrtps TRUE)
     endif()
-    if(WIN32 AND (rmw_implementation1_is_fastrtps OR rmw_implementation2_is_fastrtps))
+    if(
+      WIN32 AND
+      (rmw_implementation1_is_fastrtps OR rmw_implementation2_is_fastrtps) AND
+      NOT rmw_implementation1_is_fastrtps EQUAL rmw_implementation2_is_fastrtps
```

Comment:
Why `rmw_implementation1_is_fastrtps EQUAL rmw_implementation2_is_fastrtps` versus `rmw_implementation1_is_fastrtps AND rmw_implementation2_is_fastrtps`?

Reply:
I am not understanding your comment.

The added condition is `NOT a EQUAL b` which evaluates to false if both implementations are FastRTPS (or not FastRTPS which doesn't matter since the condition one line above is false in that case anyway).